cupboard_app/views.py

```diff
@@ -698,3 +702,69 @@ def list(self, request: Request) -> Response:
         all_measurements = get_all_measurements()
         serializer = MeasurementSerializer(all_measurements, many=True)
         return Response(serializer.data)
+
+
+@extend_schema(tags=['CustomIngredients'])
+class CustomIngredientsViewSet(viewsets.ViewSet):
+    MISSING_USER_INFO = 'User missing. Unable to create new custom ingredient.'
+
+    @extend_schema(
+        request=None,
+        responses={
+            200: CustomIngredientSerializer,
+            400: MessageSerializer,
+            401: auth_failed_response,
+        },
+        examples=[
+            OpenApiExample(
+                name='Custom Ingredient Created/Deleted.',
+                value={'username': 'teacup', 'name': 'Beef', 'type': 'Meat'},
+                status_codes=[200]
+            ),
+            OpenApiExample(
+                name='Unable to Create/Delete Custom Ingredient',
+                value={'message': MISSING_USER_INFO},
+                status_codes=[400]
+            )
+        ]
+    )
+    def create(self, request: Request) -> Response:
+        """
+        Creates a custom ingredient for the user.
+        Returns the custom ingredient object.
+        """
+        # Extract username from the access token
+        username = get_auth_username_from_payload(request=request)
+        body = request.data
+        user = get_user(username)
+        if username and user:
```

Comment:
This `user = get_user` and `if user` is not needed because this check is already done in the `create_custom_ingredient` query with the get for user happening. The get already raises an exception if the username doesn't exist in the database.